### DISCUSSION TITLE
Free VNC recommendations for Headless Windows Server?

### DISCUSSION THREAD

**Discussion by sparksandshadows** [apohl79/audiogridder#570] [General]
TightVNC is not working. Anyone have any recommendations?

**Comment by jaacMusic** [comment]
on my AG server (Windows 10), I use UltraVNC (uvnc.com) and a virtual video driver (USB Mobile Monitor Indirect Display Adapter, usbmobilemonitor.com), this allows the last one, allows various display resolutions without having a physical video card or monitor plug into the AG server.

**Reply by sparksandshadows** [reply]
Interesting, I might be doing something wrong with the virtual video driver, I'm not able to get it to run properly..

**Reply by jaacMusic** [reply]
I have to create a .bat file and added it to the windows start-up so it would load the driver every time. Maybe there is a better way, but for now, I only have to wait a few seconds before I remote to my AG server via the UltraVNC viewer.

**Reply by sparksandshadows** [reply]
I'm a bit of a noob, with this sort of thing--I know how to write and run a .bat, but which file exactly runs the driver correctly?

**Reply by jaacMusic** [reply]
The .bat file came with the drivers for installing the driver. In my case, I run it every time I start the AG server machine because for some reason it gets uninstalled after a reboot.
Here is the content of the "usbmmidd.bat" file

@cd /d "%~dp0"

@goto %PROCESSOR_ARCHITECTURE%
@exit

:AMD64
@cmd /c deviceinstaller64.exe install usbmmidd.inf usbmmidd
@cmd /c deviceinstaller64.exe enableidd 1
@goto end

:x86
@cmd /c deviceinstaller.exe install usbmmidd.inf usbmmidd
@cmd /c deviceinstaller.exe enableidd 1

:end

![image](https://user-images.githubusercontent.com/68957086/119181291-5ebbbd00-ba3f-11eb-8b63-164cd903ce28.png)

**Reply by sparksandshadows** [reply]
gotchya! that makes sense: I might still be missing a step, because I think I've launched the usb monitor but still don't have it working:
![image](https://user-images.githubusercontent.com/82538597/119186470-ffed4800-ba2c-11eb-9456-bbbe76a91dd5.png)

**Reply by sparksandshadows** [reply]
Aha! It works!!! I forgot to start the UVNC service, thank you!!! Brilliant!

**Reply by jaacMusic** [reply]
I'm glad I was able to help you!

**Comment by CaBuffalo** [comment]
Wow, thanks, Jaacmusic.
